server/app/controllers/applicant/RedirectController.java

```diff
@@ -107,32 +111,35 @@ public CompletionStage<Result> programBySlug(Http.Request request, String progra
                         // for this program, redirect to program version associated
                         // with that application if so.
                         if (programForExistingApplication.isPresent()) {
+                          long programId = programForExistingApplication.get().id();
                           return CompletableFuture.completedFuture(
-                              redirect(
-                                      controllers.applicant.routes.ApplicantProgramReviewController
-                                          .review(
-                                              applicantId,
-                                              programForExistingApplication.get().id()))
-                                  .flashing("redirected-from-program-slug", programSlug));
+                              goToActiveProgram(programId, applicantId, programSlug, request));
+                        } else {
+                          return programService
+                              .getActiveProgramDefinitionAsync(programSlug)
+                              .thenApply(
+                                  activeProgramDefinition ->
+                                      goToActiveProgram(
+                                          activeProgramDefinition.id(),
+                                          applicantId,
+                                          programSlug,
+                                          request));
                         }
-
-                        return redirectToActiveProgram(applicantId, programSlug);
                       },
                       httpContext.current());
             },
             httpContext.current());
   }
 
-  private CompletionStage<Result> redirectToActiveProgram(long applicantId, String programSlug) {
-    return programService
-        .getActiveProgramDefinitionAsync(programSlug)
-        .thenApplyAsync(
-            (activeProgramDefinition) ->
-                redirect(
-                        controllers.applicant.routes.ApplicantProgramReviewController.review(
-                            applicantId, activeProgramDefinition.id()))
-                    .flashing("redirected-from-program-slug", programSlug),
-            httpContext.current());
+  private Result goToActiveProgram(
```

Comment:
This method no longer takes you to the active program and should be renamed

Reply:
Renamed to `redirectToReviewPage`